### ISSUE TITLE
Scopy: Logic Analyzer no longer shows decoded data at changed baud rate

### ISSUE THREAD

**Issue by SgoSkz** [analogdevicesinc/scopy#1377]
**Environment:**
 - OS: Windows 10
 - FW Version: v0.25
 - SW Version: v1.4.1-f4beeb1
 
**Describe the bug**
Logic Analyzer no longer shows the decoded data for UART when changing baud rate

**To Reproduce**
1. Go to "Logic Analyzer" in Scopy
2. Set UART to respective pins
3. Set single capture
4. See decoded data
5. Change baud rate
6. See error

**Expected behavior**
Decoded data shows up

**Screenshots**
![image](https://user-images.githubusercontent.com/12992578/198785673-c09bc6a9-dabf-47fe-ad7e-c3e2faf484fd.png)
115200 (default) baud rate
![image](https://user-images.githubusercontent.com/12992578/198785821-d167e2a8-666a-4ffe-8462-d18c55cf2169.png)
38400 (changed) baud rate

**Additional context**
Scopy.ini
https://pastehub.net/fdc83de4341

**Comment by SgoSkz** [comment]
Updating firmware did not fix the issue.
New FW Version: v0.29

**Comment by adisuciu** [comment]
I don't think it should decode though. To me It looks like your signal is at 115200 baudrate, not at 38400 ..

**Comment by SgoSkz** [comment]
You are absolutely correct, 115200 is the correct baudrate, but when I change it back to 115200 from 38400, the decoded messages still dont show up. I'll add a screenshot soon

**Comment by SgoSkz** [comment]
![image](https://user-images.githubusercontent.com/12992578/199877006-09d1fab5-7b5d-4eea-8312-09b394de3cd7.png)

**Comment by jjmoreno1163** [comment]
Hi, I have the same issue, when changing to any baudrate value, decoding doesn't work.

**Comment by adisuciu** [comment]
The solution here would be to trigger another decoding when one of the decoding parameters changes if I understand correctly. It should do this even when the acquisition stopped.

**Comment by SgoSkz** [comment]
Could you elaborate a little more on that? From what I'm getting, you mean to scan for another message and it should decode properly? That is not the case for my situation, I will provide screenshots when I get the time.

**Comment by adisuciu** [comment]
Well it seems to me that there are no new acquisitions in betwseen changing baud rates.
Decoding only seems to work upon new acquisition, not upon changing decoding parameters. My suggestion is to either decode upon changing decoding parameters, or manually trigerring a new decode (via a button).

**Comment by SgoSkz** [comment]
This is an active 9600 baud rate scan with no decode messages at 9600 decode value for the logic analyzer.
![image](https://user-images.githubusercontent.com/12992578/203146225-9b47bf68-1d46-4877-a441-cd8271d850ed.png)
Changing any respective trigger settings for both DIO and extern trigger does not cause decoded messages to appear

**Comment by adisuciu** [comment]
I just tested and it works just fine for me .. 
![image](https://user-images.githubusercontent.com/8676912/203267761-a74e21df-ea0e-4df3-bebc-4d34ed5a9603.png)
Can you try removing your ini files ? https://wiki.analog.com/university/tools/m2k/scopy-troubleshooting?s[]=scopy&s[]=troubleshooting#reinstall_ini_files

**Comment by SgoSkz** [comment]
unfortunately, deleting the ini files in %AppData%/ADI does not solve the issue for me. Is the version you are testing also v1.4.1?

![image](https://user-images.githubusercontent.com/12992578/203329332-6221fe89-dc60-49c1-b9dc-1a52c8a1f667.png)
![image](https://user-images.githubusercontent.com/12992578/203329385-2e78dbd1-b5d1-4d5c-8b4a-5a60c917d893.png)

**Comment by adisuciu** [comment]
I reproduced this. I used v1.4.0 where things seem to work just fine. I will take a look.

**Comment by pitsche** [comment]
I found a workaround.
Chanche the default in C:\Program Files\Analog Devices\Scopy\decoders\uart\pd.py to the value you need.
Restart Scopy and that's it.

**Comment by molitoris** [comment]
The logic analyzer does no longer shows the decoded data for UART after changing the baud rate

OS: Windows 11
SW Version: v1.4.1-f4beeb1

**Comment by andrei47w** [comment]
This bug was fixed for all platforms on the main branch. Will be available in the next release.
